ddtrace/opentelemetry/span.go

```diff
@@ -142,26 +165,35 @@ func (s *span) SetStatus(code otelcodes.Code, description string) {
 // SetAttributes sets the key-value pairs as tags on the span.
 // Every value is propagated as an interface.
 func (s *span) SetAttributes(kv ...attribute.KeyValue) {
-	for _, attr := range kv {
-		s.SetTag(toSpecialAttributes(string(attr.Key), attr.Value))
+	for _, kv := range kv {
+		if k, v := toSpecialAttributes(string(kv.Key), kv.Value); k != "" {
+			s.attributes[k] = v
+		}
 	}
 }
 
 // toSpecialAttributes recognizes a set of span attributes that have a special meaning.
 // These tags should supersede other values.
 func toSpecialAttributes(k string, v attribute.Value) (string, interface{}) {
 	switch k {
-	case "operation.name":
-		return ext.SpanName, v.AsString()
-	case "service.name":
-		return ext.ServiceName, v.AsString()
-	case "resource.name":
-		return ext.ResourceName, v.AsString()
-	case "span.type":
-		return ext.SpanType, v.AsString()
+	case operationNameKey:
+		if ops := strings.ToLower(v.AsString()); ops != "" {
+			return ext.SpanName, strings.ToLower(v.AsString())
+		}
+		// ignoring non-string values
+		return "", nil
+	//	todo check if we need to remap these as they are the same
+	//case "service.name":
+	//	return ext.ServiceName, v.AsString()
+	//case "resource.name":
+	//	return ext.ResourceName, v.AsString()
+	//case "span.type":
+	//	return ext.SpanType, v.AsString()
```

Comment:
Still a TODO?

Reply:
Nope, removed the commented block as they are the same.